### PR TITLE
MGMT-10921: Ensure that API DNS validation checks for resolution to a master.

### PR DESCRIPTION
**Pull request by paul-maidment**
This commit adds a check to ensure that when validating the DNS resolution of the API address that none of these addresses are matched to a current master as this is the incorrect API address.

## List all the issues related to this PR

- [ ] New Feature 
- [ ] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Assignees

/cc @
/cc @

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] Reviewers have been listed
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: internal/host/validator.go [hunk 1]
```diff
@@ -1305,6 +1305,22 @@ func checkDomainNameResolution(c *validationContext, domainName string) Validati
 	for _, domain := range response.Resolutions {
 		if domain.DomainName != nil && *domain.DomainName == domainName {
 			if len(domain.IPV4Addresses) != 0 || len(domain.IPV6Addresses) != 0 {
+				for _,host := range c.cluster.Hosts{
+					ipV4Addresses, ipV6Addresses, err := network.GetHostIPAddresses(host)
+					if (err != nil) {
```
Comment (ori-amizur) [inline internal/host/validator.go:1310]: Please remove parenthesis

File: internal/host/validator.go [hunk 2]
```diff
@@ -1305,6 +1305,22 @@ func checkDomainNameResolution(c *validationContext, domainName string) Validati
 	for _, domain := range response.Resolutions {
 		if domain.DomainName != nil && *domain.DomainName == domainName {
 			if len(domain.IPV4Addresses) != 0 || len(domain.IPV6Addresses) != 0 {
+				for _,host := range c.cluster.Hosts{
```
Comment (ori-amizur) [inline internal/host/validator.go:1308]: Why in host validation, we are iterating over all the hosts ? I think we need to check only the current host.
Also please use the already parsed inventory

Comment (paul-maidment) [reply internal/host/validator.go:1308]: We are iterating over the cluster hosts as the intention is to check whether or not the currently resolved address is actually one of the cluster hosts. If this is the case then the aim of the ticket is to reject that host.

File: internal/host/validator.go [hunk 3]
```diff
@@ -1320,6 +1346,12 @@ func printIsDomainNameResolvedCorrectly(c *validationContext, status ValidationS
 		}
 		return fmt.Sprintf("Domain name resolution was successful for domain %s", domainName)
 	case ValidationFailure:
+		domain, err := getFirstMatchingResolution(c, domainName)
+		if err == nil && domain != nil {
```
Comment (ori-amizur) [inline internal/host/validator.go:1350]: Is the case err != nil handled ?

File: internal/host/hosts_suite_test.go [hunk 4]
```diff
@@ -11,6 +12,7 @@ import (
 func TestHost(t *testing.T) {
 	RegisterFailHandler(Fail)
 	common.InitializeDBTest()
+	time.Sleep(10 * time.Second)
```
Comment (tsorya) [inline internal/host/hosts_suite_test.go:15]: why we need sleep? we should not have it

Comment (paul-maidment) [reply internal/host/hosts_suite_test.go]: Oops! Left some debug stuff in...deleted!

**Pull request merged**